stdlib/test/collections/test_list.mojo

```diff
@@ -84,6 +84,34 @@ def test_list():
     assert_equal(0, len(list))
     assert_equal(2, list.capacity)
 
+    # Test pop with index
+    for i in range(6):
+        list.append(i)
+    
+    # try poping from index 3 for 3 times
+    for i in range(3, 6):
+        assert_equal(i, list.pop(3))
```

Comment:
Pop 3 times, it hides the order of pop out. You can consider popping once to check instead of popping 3 times to check the remaining members.